src/main/kotlin/com/stsf/globalbackend/controllers/HomeworkBallController.kt

```diff
@@ -0,0 +1,62 @@
+package com.stsf.globalbackend.controllers
+
+import com.stsf.globalbackend.exceptions.InsufficientPermissionsException
+import com.stsf.globalbackend.models.HomeworkBall
+import com.stsf.globalbackend.repositories.HomeworkBallRepository
+import com.stsf.globalbackend.repositories.HomeworkRepository
+import com.stsf.globalbackend.request.GenericResponse
+import com.stsf.globalbackend.services.AuthenticationService
+import com.stsf.globalbackend.services.HomeworkBallService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.bind.annotation.*
+
+
+// Maybe work on the naming
+@RestController
+@RequestMapping("/api/homework/balls")
+class HomeworkBallController (
+    @Autowired
+    private val auth: AuthenticationService,
+    @Autowired
+    private val ballService: HomeworkBallService,
+    ) {
+
+    @PutMapping("/")
+    fun addBall(@RequestHeader token: String, @RequestParam homeworkId: Long): GenericResponse<HomeworkBall> {
+        val authenticatedUser = auth.getUserByToken(token)
+        return GenericResponse(ballService.addBall(authenticatedUser.id, homeworkId))
+
+    }
+
+    @DeleteMapping("/")
+    fun deleteBall(@RequestHeader token: String, @RequestParam ballId: Long) {
```

Comment:
I think it would be better if this mapping took homeworkId as a parameter (instead of ballId)

Reply:
There is a function that does the same that you are proposing, so im thinking of just renaming it
https://github.com/7274-dev/AdventnaVyzva-GlobalBackend/blob/9246d44c81b112b504b6e4e785f5c319b3890688/src/main/kotlin/com/stsf/globalbackend/controllers/HomeworkBallController.kt#L44-L54